src/components/donation/Steps/Summary/Summary.tsx

```diff
@@ -37,7 +37,11 @@ export default function Summary({
       Amount={Amount}
       amount={amount}
       splitLiq={liquidSplitPct}
-      onBack={() => dispatch(setStep("tip"))}
+      onBack={() =>
+        recipient.hide_bg_tip
+          ? dispatch(setStep("splits"))
+          : dispatch(setStep("tip"))
+      }
       tip={
```

Comment:
The current setup requires this conditional logic or more generally, it requires devs to *know* what is the next and previous step for each component. 
Maybe it would be more practical to replace `setStep` with something like `goToNextStep` & `goToPreviousStep`, making those actions' responsibility to decide which step to go to, removing the need from devs to *remember* which step to navigate to. 
@ap-justin @SovereignAndrey let me know what you think.

Reply:
> it requires devs to know what is the next and previous step for each component.

the implementation of 
> with something like goToNextStep & goToPreviousStep, 

still requires knowing which step to set the state

Reply:
> the implementation of

Not sure I understand this, did you mean to write something more here?

> still requires knowing which step to set the state

Reducer has access to `state.step`, so these actions could be written like:
```ts
    goToNextStep: (state) => {
      if (state.step === "init") {
        state.step = "donate-form";
      } else if (...) {
      ...
    },
```
Data would be set as is currently the case (`setSplit`, `setDetails`, etc.).

Reply:
>The current setup requires this conditional logic or more generally, it requires devs to know what is the next and previous step for each component.

> still requires knowing which step to set the state

In your example, dev still has to know that the next step to `init` is `donate-form`, only difference is that this is handled in the reducer 
```js
if (state.step === "init") {
        state.step = "donate-form";
      }
```
by handling this on the dispatch, caller already has context (current step)
```
//dispatcher is in splits step (context)
--> continue: tip if enabled
<-- back: donate-form

```
contrary to handling this in the reducer, additional checks required  to know what the current step first before going to the next
```js

switch(state.currentStep){
 case "form" --> split
 case "split": --> tip if not skipped
 case "tip": --> summary
 case "summary" --> checkout 
}
```
same goes for going back
```js
switch(state.currentStep){
}
```

Reply:
> In your example, dev still has to know that the next step to init is donate-form, only difference is that this is handled in the reducer

That is true, but the difference is that the step flow is clearly defined in a single place, making it unnecessary to "remember" which step goes next/before.

> contrary to handling this in the reducer, additional checks required to know what the current step first before going to the next

This is also true, luckily with our current requirement all the necessary checks can be performed with just `state`, as it contains all the data necessary.

Anyway this was just an idea, we can table this as it's not important.